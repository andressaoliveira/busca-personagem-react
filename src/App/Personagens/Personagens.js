import React, { Component } from "react"
import PropTypes from "prop-types"
import Personagem from './Personagem'

import './Personagens.scss';

class Personagens extends Component {
  constructor(props) {
    super(props)
    this.state = {
      listaPersonagens: null,
      valorBusca: ""
    };
  }

  componentDidMount = async () => {
    const { listarPersonagens, personagens } = this.props;
    if (!personagens.loading) {
      await listarPersonagens();
    }
    this.setState({
      listaPersonagens: personagens.data
    });
  };

  handleChange(valor) {
    const { personagens } = this.props;

    this.setState({ valorBusca: valor });
    const listaPersonagensFiltrada = personagens.data.filter(p => p.name.toUpperCase().includes(valor.toUpperCase()));
    this.setState({ listaPersonagens: listaPersonagensFiltrada });
  }

  render() {
    const { listaPersonagens, valorBusca } = this.state;
    return (
      <div className="Personagens">
        <div className="cabecalho">
          <img
            src="https://yt3.ggpht.com/a/AGF-l79rcSfCz8vufHWw5DP4cTMYGvBpu135UlyfWw=s900-c-k-c0xffffffff-no-rj-mo"
            width="150"
            height="150"
            alt="logo" />

          <div className="busca">
            <input
              id="input-busca"
              type="search"
              value={valorBusca}
              onChange={(value) => this.handleChange(value.target.value)}
              placeholder="Pesquise os personagens"
            />
          </div>
        </div>
        <div className="lista-personagens">
          <div className="titulo">
            {valorBusca ? `VocÃª pesquisou por "${valorBusca}"` : 'Personagens'}
          </div>
          <div className="lista">
            {listaPersonagens && listaPersonagens.map(
              personagem => <Personagem personagem={personagem} key={personagem.char_id} />
            )}
          </div>
        </div>
      </div>
    )
  }
}

Personagens.propTypes = {
  listarPersonagens: PropTypes.func.isRequired,
  personagens: PropTypes.object,
}

Personagens.defaultProps = {
  personagens: {}
}

export default Personagens
