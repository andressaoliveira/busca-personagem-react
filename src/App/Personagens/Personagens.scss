.Personagens {
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 0 100px;

  .cabecalho{
    display: flex;
    justify-content: space-between;
    align-items: center;
    //background-image: url('../../Img/lupa-branca.png');
    input
    {
      background: url('../../Img/lupa-branca.png') no-repeat 212px center;
      color: white;
      // background-color: rgba(0,0,0,0.1);
      background-size: 20px;
      border: none;
      border-bottom: 1px solid rgba(255,255,255,0.5);
      padding-top: 2px;
      text-indent: 8px;
      width: 240px;
      height: 39px;
    }

  }
.lista-personagens{
  color: white;
  .titulo{
    font-size: 20px;
    align-items: flex-start;
    display: flex;
    padding-bottom: 15px;
  }
  .lista{
    font-size: 10px;
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    .personagem{
      width: 250px;
      height: 350px;
      padding: 10px 5px;

      .dados{
        box-shadow: 0px -50px 36px #000 inset;
        width: 250px;
        height: 350px;
        position: relative;
        bottom: 353px;
        display: flex;
        flex-direction: column;
        justify-content: space-between;

        .status{
          width: 100%;
          justify-content: flex-end;
          display: flex;
          .etiqueta{
            background-color: green;
            width: 100px;
            height: 30px;
            display: flex;
            justify-content: center;
            align-items: center;
            position: relative;
            right: 10px;
            top: 10px;
          }
        }
        .info{
          justify-content: flex-end;
          display: flex;
          flex-direction: column;
          padding-left: 10px;
          .nome{
            display: flex;
            font-size: 16px;
            font-weight: 500;
          }
          .data-nascimento{
            display: flex;
          }
          .ocupacao{
            display: flex;
            text-align: left;
          }
        }
      }
    }
  }
}
}
