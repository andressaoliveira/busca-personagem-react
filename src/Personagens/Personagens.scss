.Personagens {
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 0 100px;

  .cabecalho {
    display: flex;
    justify-content: space-between;
    align-items: center;

    //background-image: url('../../Img/lupa-branca.png');
    input {
      background: url('../Img/lupa-branca.png') no-repeat 212px center;
      color: white;
      // background-color: rgba(0,0,0,0.1);
      background-size: 20px;
      border: none;
      border-bottom: 1px solid rgba(255, 255, 255, 0.5);
      padding-top: 2px;
      text-indent: 8px;
      width: 240px;
      height: 39px;
    }

  }

  .lista-personagens {
    color: white;

    .titulo {
      font-size: 20px;
      align-items: flex-start;
      display: flex;
      padding-bottom: 15px;
    }

    .lista {
      font-size: 10px;
      display: flex;
      flex-wrap: wrap;
      justify-content: space-between;
    }
  }
}