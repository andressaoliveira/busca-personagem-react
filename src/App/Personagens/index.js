import { connect } from "react-redux";
import Personagens from './Personagens';
import { listarPersonagens } from '../../Infraestructure/Ducks/PersonagensDuck';

export const mapStateToProps = state => {
    const { personagens } = state;
    return { personagens };
};

export const mapDispatchToProps = dispatch => ({
    listarPersonagens: () => {
        dispatch(listarPersonagens())
    },
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Personagens);