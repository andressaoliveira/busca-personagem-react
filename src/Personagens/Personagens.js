import React, { useEffect, useState } from "react"
import { useDispatch, useSelector } from "react-redux"

import Personagem from '../Personagem/Personagem'
import { getPersonagens } from '../redux/personagens/slice'

import './Personagens.scss';

const Personagens = () => {
  const dispatch = useDispatch();
  const { data } = useSelector((state) => state.personagensReducer)

  const [personagens, setPersonagens] = useState([]);
  const [valorBusca, setValorBusca] = useState('');

  useEffect(() => {
    dispatch(getPersonagens());
  }, [dispatch])

  useEffect(() => {
    setPersonagens(data);
  }, [data])

  function handleChange(valor) {
    const listaPersonagensFiltrada = data.filter(p => p.name.toUpperCase().includes(valor.toUpperCase()));
    setPersonagens(listaPersonagensFiltrada);
    setValorBusca(valor)
  }

  return (
    <div className="Personagens">
      <div className="cabecalho">

        <div className="busca">
          <input
            id="input-busca"
            type="search"
            value={valorBusca}
            onChange={(value) => handleChange(value.target.value)}
            placeholder="Pesquise os personagens"
          />
        </div>
      </div>
      <div className="lista-personagens">
        <div className="titulo">
          {valorBusca ? `VocÃª pesquisou por "${valorBusca}"` : 'Personagens'}
        </div>
        <div className="lista">
          {personagens && personagens.map(
            personagem => <Personagem personagem={personagem} key={personagem.char_id} />
          )}
        </div>
      </div>
    </div>
  )
}
export default Personagens;