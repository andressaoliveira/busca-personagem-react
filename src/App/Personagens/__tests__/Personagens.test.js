import expect from "expect";
import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import { mapStateToProps, mapDispatchToProps } from "../index"
import Personagens from "../index";

configure({ adapter: new Adapter() })

const mockStore = configureMockStore();
const store = mockStore({})
describe("Personagens", () => {
  function setup() {
    const props = {
      personagens: {
        data: [
          {
            char_id: 18,
            name: "Brandon Mayhew",
            birthday: "Unknown",
            occupation: [
              "Low-level meth distributer"
            ],
            img: "https://images.amcnetworks.com/amc.com/wp-content/uploads/2015/04/cast_bb_700x1000_badger-lg.jpg",
            status: "Alive",
            nickname: "Badger",
            appearance: [
              1,
              2,
              3,
              4,
              5
            ],
            portrayed: "Matt L. Jones"
          },
          {
            char_id: 19,
            name: "Huell Babineaux",
            birthday: "Unknown",
            occupation: [
              "Bodyguard"
            ],
            img: "https://vignette.wikia.nocookie.net/breakingbad/images/c/c1/4x11_-_Huell.png/revision/latest?cb=20130913100459&path-prefix=es",
            status: "Alive",
            nickname: "Huell",
            appearance: [
              4,
              5
            ],
            portrayed: "Lavell Crawford"
          },
          {
            char_id: 20,
            name: "Steven Gomez",
            birthday: "Unknown",
            occupation: [
              "DEA Agent"
            ],
            img: "https://images.amcnetworks.com/amc.com/wp-content/uploads/2015/04/cast_bb_700x1000_steven-gomez-lg.jpg",
            status: "Deceased",
            nickname: "Gomie",
            appearance: [
              1,
              2,
              3,
              4,
              5
            ],
            portrayed: "Steven Michael Quezada"
          },
        ]
      }
    };
    return shallow(
      <Provider store={store}>
        <Personagens {...props} />
      </Provider>);
  }

  const wrapper = setup();

  it("deve montar o componente sem deixar de funcionar", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Testes do index", () => {
    const funct = { personagens: {}}
    const dispatch = jest.fn()
    expect(mapStateToProps(funct)).toEqual(funct)
    mapDispatchToProps(dispatch).listarPersonagens()
    expect(dispatch).toHaveBeenCalled()
  })
});
