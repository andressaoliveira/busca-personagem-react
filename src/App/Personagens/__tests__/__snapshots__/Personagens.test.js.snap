// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Personagens deve montar o componente sem deixar de funcionar 1`] = `ShallowWrapper {}`;
